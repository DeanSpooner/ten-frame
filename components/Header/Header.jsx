import React from "react";
import { View, Text } from "react-native";
import HamburgerIcon from "../HamburgerIcon/HamburgerIcon";
import headerStyles from "./Header.styles";

const Header = () => {
  return (
    <View style={headerStyles.container}>
      <HamburgerIcon />
      <Text style={headerStyles.title}>Ten Frame</Text>
      <View style={{ flex: 1, margin: 10 }} />
    </View>
  );
};

export default Header;
