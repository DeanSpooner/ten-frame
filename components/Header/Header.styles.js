import { StyleSheet } from "react-native";

export default headerStyles = StyleSheet.create({
  container: {
    flexDirection: "row",
    width: "100%",
    height: 100,
    backgroundColor: "green",
    position: "absolute",
    top: 0,
    alignItems: "flex-end",
    justifyContent: "center",
    paddingBottom: 5,
    flex: 1,
  },
  title: {
    color: "#fff",
    fontSize: 40,
    textAlign: "center",
  },
});
