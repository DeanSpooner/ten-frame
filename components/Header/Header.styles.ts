import { StyleSheet } from "react-native";

const headerStyles = StyleSheet.create({
  outerContainer: {
    flexDirection: "column",
    width: "100%",
    position: "absolute",
    top: 0,
    alignItems: "center",
    alignContent: "center",
    justifyContent: "center",
    paddingBottom: 5,
    flexWrap: "wrap",
  },
  container: {
    flexDirection: "row",
    width: "100%",
    height: 100,
    backgroundColor: "green",
    position: "absolute",
    top: 0,
    alignItems: "flex-end",
    alignContent: "flex-end",
    paddingBottom: 5,
    flexWrap: "wrap",
  },
  title: {
    color: "#fff",
    fontSize: 40,
    textAlign: "center",
  },
  menuItemContainer: {
    zIndex: 102,
    elevation: 102,
    backgroundColor: "rgba(0,255,0,0.5)",
    borderRadius: 12,
    alignItems: "center",
    marginBottom: 20,
  },
  menuItem: {
    borderRadius: 12,
    color: "#fff",
    padding: 16,
    fontSize: 30,
    textAlign: "center",
    justifyContent: "center",
  },
});

export default headerStyles;
