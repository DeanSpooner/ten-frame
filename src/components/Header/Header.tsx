import React from "react";
import { View, Text, TouchableOpacity } from "react-native";
import HamburgerIcon from "../HamburgerIcon/HamburgerIcon";
import headerStyles from "./Header.styles";

interface HeaderProps {
  setScreen: React.Dispatch<React.SetStateAction<string>>;
  setMenuDown: React.Dispatch<React.SetStateAction<boolean>>;
  menuDown: boolean;
}

const Header: React.FC<HeaderProps> = ({
  setScreen,
  setMenuDown,
  menuDown,
}) => {
  const screens: { name: string; text: string }[] = [
    { name: "home", text: "Home" },
    { name: "tenframe", text: "Ten Frame - Input" },
    { name: "tenframetouch", text: "Ten Frame - Touch" },
    { name: "tenframegame", text: "Ten Frame - Game" },
    { name: "about", text: "About" },
  ];

  return (
    <>
      <View
        style={[
          headerStyles.outerContainer,
          //@ts-ignore
          menuDown ? { height: 600, zIndex: 100, elevation: 100 } : "",
        ]}
      >
        {menuDown && (
          <View style={{ zIndex: 101, elevation: 101 }}>
            {screens.map((screen) => (
              <TouchableOpacity
                onPress={() => {
                  setScreen(screen.name);
                  setMenuDown(!menuDown);
                }}
                style={headerStyles.menuItemContainer}
                key={screen.name}
              >
                <Text style={headerStyles.menuItem}>{screen.text}</Text>
              </TouchableOpacity>
            ))}
          </View>
        )}
        <View
          style={[
            headerStyles.container,
            //@ts-ignore
            menuDown ? { height: 600, zIndex: 100, elevation: 100 } : "",
          ]}
        >
          <HamburgerIcon setMenuDown={setMenuDown} menuDown={menuDown} />
          <Text style={headerStyles.title}>Ten Frame</Text>
          <View style={{ flex: 1, margin: 10 }} />
        </View>
      </View>
    </>
  );
};

export default Header;
