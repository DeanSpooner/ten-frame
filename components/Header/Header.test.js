import React from "react";
import renderer from "react-test-renderer";
import Header from "./Header";

describe("<Header />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 1 child", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});
