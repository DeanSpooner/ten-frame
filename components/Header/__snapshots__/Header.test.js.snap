// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-end",
      "backgroundColor": "green",
      "flex": 1,
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "center",
      "paddingBottom": 5,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "margin": 10,
        "width": 30,
      }
    }
  >
    <View
      style={
        {
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 2,
          "marginTop": 8,
          "width": 35,
        }
      }
    />
    <View
      style={
        {
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 2,
          "marginTop": 8,
          "width": 35,
        }
      }
    />
    <View
      style={
        {
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 2,
          "marginTop": 8,
          "width": 35,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 40,
        "textAlign": "center",
      }
    }
  >
    Ten Frame
  </Text>
  <View
    style={
      {
        "flex": 1,
        "margin": 10,
      }
    }
  />
</View>
`;
